x-pack/plugins/endpoint/public/embeddables/resolver/view/defs.tsx

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { memo } from 'react';
+import { darken, saturate } from 'polished';
+
+import {
```

Comment:
Could you add a doc comment here that explains a little bit about your approach and design goals. Something like 'Defs are SVG elements that are placed in the DOM once and reused by other SVG elements. These defs are reused by the process nodes and edge lines. Colors are also defined here...`

Reply:
Done